controllers/secret.go

```diff
@@ -80,18 +83,60 @@ func (r *OperatorPipelineReconciler) reconcileKubeConfigSecret(meta metav1.Objec
 
 // reconcileGitHubAPISecret will ensure that the GitHub API Secret is present and up to date.
 func (r *OperatorPipelineReconciler) reconcileGitHubAPISecret(meta metav1.ObjectMeta) error {
+
+	namespaceName := types.NamespacedName{
+		Namespace: meta.Namespace,
+		Name:      meta.Name,
+	}
+
+	operatorPipeline := &certv1alpha1.OperatorPipeline{}
+	err := r.Client.Get(context.Background(), namespaceName, operatorPipeline)
+	if err != nil {
+		if k8errors.IsNotFound(err) {
+			log.Info("pipeline resource not found. Ignoring since object must be deleted")
+			return nil
+		}
+
+		log.Error(err, fmt.Sprintf("unable to retrieve pipeline resource in namespace: %s ", meta.Namespace))
+		return err
+	}
+
+	var gitHubSecretName = GITHUB_API_SECRET
+	if operatorPipeline.Spec.GitHubSecretName != "" {
+		gitHubSecretName = operatorPipeline.Spec.GitHubSecretName
+	}
+
 	key := types.NamespacedName{
 		Namespace: meta.Namespace,
-		Name:      GITHUB_API_SECRET,
+		Name:      gitHubSecretName,
 	}
 
 	secret := newSecret(key)
-	if IsObjectFound(r.Client, key, secret) {
-		log.Info("existing github api secret found")
-		return nil // Existing Secret found, do nothing...
+	if !IsObjectFound(r.Client, key, secret) {
+		return errors.New(fmt.Sprintf("github api secret not found in namespace: %s", meta.Namespace))
+	}
+
+	gitHubSecret := &corev1.Secret{}
+	err = r.Get(context.Background(), key, gitHubSecret)
```

Comment:
In most examples that I've seen, context.TODO is used rather than `.Background()`. To me, context.TODO fits better.